### ISSUE TITLE
Function argument with additional properties over function parameter type doesn't throw error

### ISSUE THREAD

**Issue by uiii** [Microsoft/TypeScript#19071]
**TypeScript Version:**  2.5.3

**Code**

```ts
interface IType {
    a: number
}

function f(p: IType) {}

const variableWithAdditionalProperty = {
    a: 5,
    b: 6
};

const variableWithMissingProperty = {
    b: 6
};

f(variableWithAdditionalProperty); // Unexpected behavior => No error

f({
    a: 5, 
    b: 6  // Expected behavior => Error: Argument of type '{ a: number; b: number; }' 
});       // is not assignable to parameter of type 'IType'. Object literal may only
          // specify known properties, and 'b' does not exist in type 'IType'.

f(variableWithMissingProperty); // Expected behavior => Error: Argument of type '{ b: number; }' 
                                // is not assignable to parameter of type 'IType'.  Property 'a' 
                                // is missing in type '{ b: number; }'.
```

Consider you have an interface declaring some properties and a function accepting a parameter of this interface type. Now you make a first function call by passing a function an object literal with all the properties of the the interface and **some additional property**. Then you assign the same object literal to a variable and make a second call by passing this variable to the function.

**Expected behavior:**
Both function calls throw an error telling the passed argument is not assignable to the interface type because of the additional property.

**Actual behavior:**
First function call (with object literal) thows an error as expected, but the second call (with variable) doesn't throw any error.

**Comment by RyanCavanaugh** [comment]
Same as #15212 - we know that `b` is unobserved in the object literal case, but have to use structural subtyping in the other case

**Comment by uiii** [comment]
Ah, I see. It makes sense.
